sycl/include/sycl/detail/generic_type_traits.hpp

```diff
@@ -378,8 +378,27 @@ template <typename T> class TryToGetPointerVecT {
   template <typename A>
   static typename TryToGetVectorT<A>::type *check(const A *);
 
+  static auto get_type() {
+    using typeVecPointer = decltype(check(T()));
+    using typeVec = std::remove_pointer_t<typeVecPointer>;
+    if constexpr (sizeof(typeVec) <= 64)
+      return typeVecPointer();
+    else {
+      using elementType = typename TryToGetElementType<typeVec>::type;
+      // The only cases where the vec size is greater than 64 is when we have a
+      // vec16 of an 8-byte type (like double, or long long)  To get around
+      // alignment issues, those are represented internally as std::array, but
+      // when interfacing with SPIR-V, we need to squint and say it is a vector
+      // extension.
+      using DT = typename DecoratedType<elementType
+                                        __attribute__((ext_vector_type(16))),
+                                        T::address_space>::type *;
+      return DT();
```

Comment:
@romanovvlad Here I'm fixing the work group copying problem. It's a bit ungainly. Also it is introducing a warning when SYCL is built ( `warning: ignoring attributes applied to dependent type ‘elementType’ without an associated declaration [-Wattributes]` ).

Wanted to get your opinion before I start tackling formatting and any more test case updates.

Reply:
Personally I would prefer trying to split 128 byte vec to two 64 byte vectors and call builtins twice.
But if we go with casting to 128 byte vector. Not sure I fully understand why it doesn't work out of the box, these helpers use `vec::vector_t`. I guess for > 64 byte types `vec::vector_t` is `std::array` which doesn't seem to be correct. If it so, wouldn't changing `vec::vector_t` for > 64 byte types to corresponding `ext_vector_type` fix the issue with async copy builtin?

Reply:
AFAICT the `vector_t` operator is not used by the helpers. They use `.get()`.

Having `vector_t` return the vector extension seems correct, but I'm not sure it matters - I don't think it has any callers.